Started implementing score functionality

Azukee opened the pull request:
I am starting to implement the score functions
(GetScores, Submit Scores etc)
If you plan on merging these builds please give credit, it would help alot <3<3<3

Commit:
Merge remote-tracking branch 'refs/remotes/Igoorx/master'

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into Igoorx/master

Commit:
Updated to latest build from Igoorx

Commit:
Started to implement score functions as well as the getscores logic.
Please if you plan on merging my builds with yours give credit, I would appreciate it <3<3

Igoorx commented:
It looks good, but as I commented at the source, in my point of view, this is not advisable because it would increase the number of requests received by Bancho, but as it does not seem to be a very big problem, and scores in bancho seems a good idea for when the person does not have an apache, I will merge, but I prefer to wait a little more to merge, as it still does nothing.

Azukee commented:
I can understand, just felt like already putting this here for the future. If you want me to do anything (if you're too lazy or don't understand it) go ahead and tell me about it

Commit:
Adding Getscores funcionality.
Osu doesn't return an error but doesn't show the score. Figure it out if you can.

Igoorx commented:
I'm too lazy to make a list of friends, if you do it would help :v
This problem that you mentioned in commit can be because you isn't setting the id in the response, and I had not noticed before, but you can get the parameters just using `context.Request.QueryString`, no need to split at all.

Azukee commented:
Oh wow I'm an idiot.
Also which ID do you mean?

Commit:
Updating of score submitting and getscores functionality.

Commit:
More Updates Yadayada

Commit:
Updated to latest push from IgoorX

Igoorx commented:
beatmapId, but you it put then took so I guess it's not that..

This is an old code that i used to test (in php)

```
2|false|648339|<?php echo $mapid; ?>|0
0
[bold:0,size:20]Artista|Titulo
9.19141
149646854|<?php echo $username; ?>|9883498|528|192|4|524|0|2|84|1|24|<?php echo $userid; ?>|1|1440523252|1
149646854|<?php echo $username; ?>|9883498|528|192|4|524|0|2|84|1|24|<?php echo $userid; ?>|1|1440523252|1
```

As you see, it is necessary to have 2 score or one white line and an score

Azukee commented:
http://i.imgur.com/KGC1Wrf.jpg
Ahahaha, wow. The dual score sending worked now. ´That took long xD. Thanks!

Commit:
GetScores functionality works

Azukee commented:
do you perhaps have the code to decrypt AES that actually works? mine doesnt work and i cant figure out why

Igoorx commented:
No, but it is not difficult to make with the help of google .. I can try to help if you provide the code used

avail commented:
If you do not have a score on the map, the first score line should be empty.

Azukee commented:
yeah i got that. sending /n is good for that
i am not working on this anymore, cant get aes to work and im too lazy to figure it out.

Igoorx commented:
It should be something like this:
http://pastebin.com/D19G5sWC

most likely with `rj.BlockSize = 256;`

Commit:
Submitting Scores now works, the only thing left that  I have to properly do is getscores.......

Commit:
Fixed GetScores and added new .sql file so everything works without problems

Azukee commented:
Update: Submitting Scores and Getting them works flawlessly now. (time is a little messed up, but meh fair enough)
(also if being forked pls give me credits)

Commit:
Updated Getscores to be mode specific c:

Commit:
Added userpanel stuff

Commit:
Adding rankings with score (pp is gonna come soon)

Commit:
Added support for bigger scores than Int32.MaxValue();
(and also some random fixes here and there c:)

Commit:
Added support for avatars which is REALLY dirty, like... REALLY FUCKING DIRTY

Commit:
Made GetScores get the username with the User id which makes name changes in the future possible without breaking the score system

Commit:
Added a default function for the avatars

Igoorx commented:
Nice, I'll get a better look in the commits later.
